src/tests/ftest/deployment/critical_integration.py

```diff
@@ -59,14 +61,16 @@ def test_passwdlessssh_versioncheck(self):
         failed_nodes = NodeSet()
         for host in self.hostlist_servers:
             daos_server_cmd = ("ssh -oNumberOfPasswordPrompts=0 {}"
-                               " 'daos_server version'".format(host))
+                               " 'daos_server version -j'".format(host))
             remote_root_access = ("ssh -oNumberOfPasswordPrompts=0 root@{}"
                                   " 'echo hello'".format(host))
             command_for_inter_node = ("clush --nostdin -S -b -w {}"
                                       " 'echo hello'".format(str(self.hostlist_servers)))
             try:
-                out = run_command(daos_server_cmd)
-                daos_server_version_list.append(out.stdout.split(b' ')[3])
+                out = json.loads((run_command(daos_server_cmd)).stdout)
+                if 'response' in out:
+                    if 'version' in out['response']:
+                        daos_server_version_list.append(out['response']['version'])
```

Comment:
I think we need to handle when `out['response']['version']` is not in the output. Otherwise, if the command fails and doesn't return a proper response, the check below for `daos_server_version_list` will silently ignore this. And the same for the dmg command. I.e.
```
        result_daos_server = (daos_server_version_list.count(daos_server_version_list[0])
                              == len(daos_server_version_list))
        result_dmg = dmg_version_list.count(dmg_version_list[0]) == len(dmg_version_list)
```
would still pass if only 1/X servers had a proper response.

We could probably just do `daos_server_version_list.append(out['response']['version'])`
and then `except (DaosTestError, CommandFailure, KeyError)` to account for an improper response